Core/src/org/sleuthkit/autopsy/guiutils/JavaFXUtils.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2017 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.guiutils;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.logging.Level;
+import javafx.scene.control.Alert;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.Dialog;
+import javafx.scene.image.Image;
+import javafx.stage.Modality;
+import javafx.stage.Stage;
+import org.openide.util.NbBundle;
+import org.sleuthkit.autopsy.coreutils.Logger;
+import org.sleuthkit.autopsy.coreutils.ThreadConfined;
+
+/**
+ * Utilities for dealing with JavaFX gui components.
+ *
+ */
+public class JavaFXUtils {
```

Comment:
This class does not feel like it is very cohesive, and I'm not enthusiastic about adding adding a showTooManyFiles method to the public API, which I personally view as a stop gap measure. I think I would rather have some code duplication here, consistent with the decomposition of the code into NBMs, then expand the API this way.

Reply:
ok I will split it up and keep it out of the API